File: CodeTest1.py
def diff_int_max(diffs):
    max_value = float('-inf')
    for num in diffs:
        if num >= max_value:
            max_value = num
    return max_value


def diff_int_min(diffs):
    min_value = float('inf')
    for num in diffs:
        if num <= min_value:
            min_value = num
    return min_value


def solution(diffs, times, limit):
    max_level = diff_int_max(diffs)
    min_level = diff_int_min(diffs)
    mid_level = (max_level + min_level) // 2
    print(mid_level)
    spend_times = 0

    while True:
        for x in range(len(diffs)):
            if x == 0:
                if diffs[x] <= mid_level:
                    spend_times = times[x]
                else:
                    spend_times = (diffs[x] - mid_level) * times[x]
            else:
                if diffs[x] <= mid_level:
                    spend_times += times[x]
                else:
                    spend_times += (diffs[x] - mid_level) * (times[x - 1] + times[x]) + times[x]

        if spend_times > limit:
            spend_times = 0
            next_level = (mid_level + max_level) // 2

            if mid_level == min_level:
                mid_level = next_level + 1
                break

            min_level = mid_level
            mid_level = next_level
            print(f"{min_level} {mid_level} {max_level}")
            continue
        else:
            spend_times = 0
            next_level = (mid_level + min_level) // 2

            if mid_level == max_level:
                mid_level = next_level
                break

            max_level = mid_level
            mid_level = next_level
            print(f"{min_level} {mid_level} {max_level}")
            continue

    print(mid_level)
    return mid_level

File: CodeTest1.cs
using System;




public class Solution {
    public int diffIntMax(int[] diffs){
        int max = int.MinValue;
        foreach (int num in diffs){
            if(num>=max){
                max=num;
            }
        }
        return max; 
    }
    
    public int diffIntMin(int[] diffs){
        int min = int.MaxValue;
        foreach (int num in diffs){
            if(num<=min){
                min=num;
            }
        }
        return min; 
    }

    public int solution(int[] diffs, int[] times, long limit) {
        int Maxlevel=diffIntMax(diffs);
        int Minlevel=diffIntMin(diffs);
        int beforelevel = 0;
        int Midlevel=(Maxlevel+Minlevel)/2;
        Console.WriteLine(Midlevel);
        long spend_times =0;
        while(true){
            for(int x =0;x<diffs.Length;x++)
            {

                if(x==0)
                {
                    if(diffs[x]<=Midlevel){
                        spend_times=times[x];
                    }
                    else{
                        spend_times=(diffs[x]-Midlevel)*(times[x]);
                    }
 
                }
                else
                {
                    if(diffs[x]<=Midlevel){
                        spend_times=spend_times+(times[x]);
                    }
                    else{
                        spend_times=spend_times+((diffs[x]-Midlevel)*(times[x-1]+times[x])+times[x]);
                    }
                }

            }
            if(spend_times>limit)
            {
                spend_times=0;
                int nextlevel=0;
                nextlevel=(Midlevel+Maxlevel)/2;
                
                if(Midlevel==Minlevel){
                    Midlevel=nextlevel+1;
                    break;
                }
                
                Minlevel=Midlevel;
                Midlevel=nextlevel;
                Console.WriteLine($"{Minlevel} {Midlevel} {Maxlevel}");
                continue;
            }
            else
            {
                spend_times=0;
                int nextlevel=0;
                
                nextlevel=(Midlevel+Minlevel)/2;



                if(Midlevel==Maxlevel){
                    Midlevel=nextlevel;
                    break;
                }
                Maxlevel=Midlevel;
                Midlevel=nextlevel;
                Console.WriteLine($"{Minlevel} {Midlevel} {Maxlevel}");
                continue;
            }
            
        }
        Console.WriteLine(Midlevel);
        return Midlevel;
    }
}